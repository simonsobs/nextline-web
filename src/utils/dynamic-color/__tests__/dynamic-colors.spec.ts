import { describe, it, expect } from "vitest";
import { useDynamicColors } from "..";

describe("useDynamicColors", () => {
  it("returns colors", () => {
    const { colors } = useDynamicColors();
    expect(colors.value).toMatchSnapshot();
  });

  it("colors are reactive", () => {
    const { colors, sourceColorHex } = useDynamicColors({
      sourceColorHex: "#F44336",
    });
    const colors1 = colors.value;
    sourceColorHex.value = "#673AB7";
    const colors2 = colors.value;
    expect(colors1).not.toEqual(colors2);
  });
});
