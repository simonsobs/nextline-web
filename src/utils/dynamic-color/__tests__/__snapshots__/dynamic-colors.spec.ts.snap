// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`useDynamicColors > returns colors 1`] = `
{
  "background": "#fdf7ff",
  "error": "#ba1a1a",
  "error-container": "#ffdad6",
  "inverse-on-surface": "#f5eff7",
  "inverse-primary": "#cfbcff",
  "inverse-surface": "#322f35",
  "neutral-palette-key-color": "#7a767d",
  "neutral-variant-palette-key-color": "#7a7582",
  "on-background": "#1d1b20",
  "on-error": "#ffffff",
  "on-error-container": "#410002",
  "on-primary": "#ffffff",
  "on-primary-container": "#ffffff",
  "on-primary-fixed": "#22005d",
  "on-primary-fixed-variant": "#4f378a",
  "on-secondary": "#ffffff",
  "on-secondary-container": "#4a4162",
  "on-secondary-fixed": "#1f1635",
  "on-secondary-fixed-variant": "#4b4263",
  "on-surface": "#1d1b20",
  "on-surface-variant": "#494551",
  "on-tertiary": "#ffffff",
  "on-tertiary-container": "#ffffff",
  "on-tertiary-fixed": "#241a00",
  "on-tertiary-fixed-variant": "#594400",
  "outline": "#7a7582",
  "outline-variant": "#cbc4d2",
  "primary": "#513a8d",
  "primary-container": "#765fb4",
  "primary-fixed": "#e9ddff",
  "primary-fixed-dim": "#cfbcff",
  "primary-palette-key-color": "#8069bf",
  "scrim": "#000000",
  "secondary": "#63597c",
  "secondary-container": "#e5d9ff",
  "secondary-fixed": "#e9ddff",
  "secondary-fixed-dim": "#cdc0e9",
  "secondary-palette-key-color": "#7c7296",
  "shadow": "#000000",
  "surface": "#fdf7ff",
  "surface-bright": "#fdf7ff",
  "surface-container": "#f2ecf4",
  "surface-container-high": "#ece6ee",
  "surface-container-highest": "#e6e0e9",
  "surface-container-low": "#f8f2fa",
  "surface-container-lowest": "#ffffff",
  "surface-dim": "#ded8e0",
  "surface-tint": "#6750a4",
  "surface-variant": "#e7e0ee",
  "tertiary": "#5d4700",
  "tertiary-container": "#876a13",
  "tertiary-fixed": "#ffdf93",
  "tertiary-fixed-dim": "#e7c365",
  "tertiary-palette-key-color": "#c9a74d",
}
`;
